File: explorer/darwin/ios/lynx_explorer/LynxExplorer/testbench/TestBenchDefaultActionCallback.m
// Copyright 2025 The Lynx Authors. All rights reserved.
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.

#import "TestBenchDefaultActionCallback.h"
#import <Lynx/LynxProviderRegistry.h>
#import <Lynx/LynxRawTextShadowNode.h>
#import <Lynx/LynxTextShadowNode.h>
#import <Lynx/LynxUIText.h>
#import <Lynx/LynxUIView.h>
#import <Lynx/LynxView.h>
#import "ExplorerModule.h"

static const int kVirtual = 1 << 2;

@interface TestBenchDefaultActionCallback ()
@end

@implementation TestBenchDefaultActionCallback

- (void)onLynxViewWillBuild:(TestBenchActionManager *)manager builder:(LynxViewBuilder *)builder {
  [builder.config registerModule:[ExplorerModule class]];
}
- (void)onLynxViewDidBuild:(LynxView *)lynxView {
}
@end

File: explorer/android/lynx_explorer/src/main/java/com/lynx/explorer/shell/TestBenchDefaultActionCallback.java
// Copyright 2025 The Lynx Authors. All rights reserved.
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.

package com.lynx.explorer.shell;

import com.lynx.devtool.testbench.TestBenchActionCallback;
import com.lynx.devtool.testbench.TestBenchActionManager;
import com.lynx.tasm.LynxGroup;
import com.lynx.tasm.LynxGroup.LynxGroupBuilder;
import com.lynx.tasm.LynxViewBuilder;

public class TestBenchDefaultActionCallback implements TestBenchActionCallback {
  @Override
  public void onLynxViewWillBuild(TestBenchActionManager manager, LynxViewBuilder builder) {
    String[] paths = manager.getTestBenchPreloadScripts();

    LynxGroup lynxGroup = manager.getLynxGroup();
    if (lynxGroup == null) {
      lynxGroup = new LynxGroupBuilder().setGroupName("testbench").setPreloadJSPaths(paths).build();
      manager.setLynxGroup(lynxGroup);
    }
    builder.setLynxGroup((LynxGroup) lynxGroup);
  }
}
